core/src/mindustry/world/blocks/logic/LogicBlock.java

```diff
@@ -485,6 +485,12 @@ public boolean onConfigureTileTapped(Building other){
             return super.onConfigureTileTapped(other);
         }
 
+        @Override
+        public double sense(LAccess sensor){
+            if(sensor == LAccess.controlling) return Groups.unit.count(u -> u.controller() instanceof LogicAI ai && ai.controller == this);
```

Comment:
Two issues with this:
- O(n) search through all units is very slow and not an acceptable solution
- This should be a global constant like `@unit`, sensing controlled count of self doesn't fit how other variables are fetched